crates/subspace-networking/src/constructor.rs

```diff
@@ -507,14 +506,18 @@ where
                 ..ConnectedPeersConfig::default()
             }
         }),
-        autonat: enable_autonat.then(|| AutonatConfig {
+        autonat: (enable_autonat && external_addresses.is_empty()).then(|| AutonatConfig {
```

Comment:
We enable autonat if requested and there are no external addresses, majority of users will not specify external addresses and when they do autonat is no longer necessary/helpful.

Reply:
When we have a user with a valid public address who doesn't specify its address as a separate parameter we make him invisible to the network. Correct?

Reply:
It makes sense. What's about the inverse situation: we set external address but since we set subspace_networking `KademliaMode` to Dynamic and we have libp2p Kademlia mode Client  - we don't toggle it to Server state.

Reply:
It is possible, I want to remove autonat and kademlia mode options completely, should I do it in this PR or separate?